packages/syft/src/syft/core/adp/vectorized_publish.py

```diff
@@ -155,7 +157,19 @@ def publish(
         # Step 2: Calculate the epsilon spend for this query
 
         # rdp_constant = all terms in Theorem. 2.7 or 2.8 of https://arxiv.org/abs/2008.11193 EXCEPT alpha
-        rdp_constants = compute_rdp_constant(rdp_params, private=private)
+        if any(np.isnan(l2_norms)):
```

Comment:
@teo-milea nice! Could we extract this into a function which takes the input and then raises a custom exception so theres no need for indenting and we can catch that specific exception in the outer part of the function?